pkg/k8s/factory_functions.go

```diff
@@ -423,24 +439,54 @@ func ConvertToCNP(obj interface{}) interface{} {
 		}
 		*concreteObj = cilium_v2.CiliumNetworkPolicy{}
 		return cnp
-	case cache.DeletedFinalStateUnknown:
-		cnp, ok := concreteObj.Obj.(*cilium_v2.CiliumNetworkPolicy)
-		if !ok {
-			return obj
+
```

Comment:
Please create a new function dedicated for the conversion of `CiliumClusterwideNetworkPolicy` to `types.SlimpCNP`